pkg/controllers/state/cluster.go

```diff
@@ -221,11 +221,10 @@ func (c *Cluster) UpdateNodeClaim(nodeClaim *v1beta1.NodeClaim) {
 	c.mu.Lock()
 	defer c.mu.Unlock()
 
-	if nodeClaim.Status.ProviderID == "" {
-		return // We can't reconcile nodeclaims that don't yet have provider ids
+	if nodeClaim.Status.ProviderID != "" {
```

Comment:
Add a comment here on why we are only doing this when nodeClaim status is non-empty